### PR TITLE
Implement application

### PR DESCRIPTION
**Pull request by vladzhorov**

### CODE REVIEW

**Review by pavelGagarin0870 — APPROVED** [review]

File: ServiceFinder.OrderService/ServiceFinder.OrderService.Application/ApplicationExtension.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+using ServiceFinder.OrderService.Application.Messaging;
+using ServiceFinder.OrderService.Application.Messaging.RabbitMQConfigurations;
+
+namespace ServiceFinder.OrderService.Application
+{
+    public static class ApplicationExtension
+    {
+        public static void AddApplication(this IServiceCollection services, IConfiguration configuration)
+        {
+            services.Configure<RabbitMQConfiguration>(options => configuration.GetSection("RabbitMQ").Bind(options));
```
Comment (moltenessence) [inline ServiceFinder.OrderService/ServiceFinder.OrderService.Application/ApplicationExtension.cs:16]: move section name to constants

File: ServiceFinder.OrderService/ServiceFinder.OrderService.Application/ApplicationExtension.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+                channel.ExchangeDeclare(exchange: options.Exchange!.Name, type: options.Exchange.Type);
+                channel.QueueDeclare(queue: options.Queue!.Name,
+                                     durable: false,
+                                     exclusive: false,
+                                     autoDelete: false,
+                                     arguments: null);
+                channel.QueueBind(queue: options.Queue.Name,
+                                  exchange: options.Exchange.Name,
+                                  routingKey: options.Queue.RoutingKey);
```
Comment (moltenessence) [inline ServiceFinder.OrderService/ServiceFinder.OrderService.Application/ApplicationExtension.cs:37]: I don't understand what's the queue is from its naming

File: ServiceFinder.OrderService/ServiceFinder.OrderService.Application/EventHandlers/OrderRequestStatusChangedEventHandler.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
+﻿using ServiceFinder.OrderService.Application.Messaging;
+using ServiceFinder.OrderService.Domain.Events;
+using System.Text.Json;
+
+namespace ServiceFinder.OrderService.Application.EventHandlers
+{
+    public class OrderRequestStatusChangedEventHandler
```
Comment (moltenessence) [inline ServiceFinder.OrderService/ServiceFinder.OrderService.Application/EventHandlers/OrderRequestStatusChangedEventHandler.cs:7]: Why don't you use interface?

File: ServiceFinder.OrderService/ServiceFinder.OrderService.Application/Mapper/OrderMappingProfile.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+using ServiceFinder.OrderService.Domain.Models;
+
+namespace ServiceFinder.OrderService.Application.Mapper
+{
+    public class OrderMappingProfile : Profile
+    {
+        public OrderMappingProfile()
+        {
+            CreateMap<Order, OrderDto>();
+            CreateMap<OrderDto, Order>();
```
Comment (moltenessence) [inline ServiceFinder.OrderService/ServiceFinder.OrderService.Application/Mapper/OrderMappingProfile.cs:12]: .ReverseMap()

File: ServiceFinder.OrderService/ServiceFinder.OrderService.Application/Mapper/OrderMappingProfile.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+            CreateMap<OrderRequest, OrderRequestDto>();
+            CreateMap<OrderRequestDto, OrderRequest>();
```
Comment (moltenessence) [inline ServiceFinder.OrderService/ServiceFinder.OrderService.Application/Mapper/OrderMappingProfile.cs:15]: same

File: ServiceFinder.OrderService/ServiceFinder.OrderService.Application/Services/OrderRequestAppService.cs [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        {
+            var orderRequest = _mapper.Map<OrderRequest>(orderRequestDTO);
+            await _orderRequestService.CreateOrderRequestAsync(orderRequest, cancellationToken);
+            return _mapper.Map<OrderRequestDto>(orderRequest);
+        }
+
+        public async Task UpdateOrderRequestStatusAsync(Guid orderRequestId, OrderRequestStatus newStatus, CancellationToken cancellationToken)
+        {
+            await _orderRequestService.UpdateOrderRequestStatusAsync(orderRequestId, newStatus, cancellationToken);
+        }
```
Comment (moltenessence) [inline ServiceFinder.OrderService/ServiceFinder.OrderService.Application/Services/OrderRequestAppService.cs:33]: Avoid state machine generation under the hood

**Review by moltenessence — CHANGES_REQUESTED** [review]

**Review by moltenessence — APPROVED** [review]

File: ServiceFinder.OrderService/ServiceFinder.OrderService.Application/ApplicationExtension.cs [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+            services.AddScoped<IMessagePublisher, MessagePublisher>();
+        }
+
+        public static void AddApplication(this IServiceCollection services)
+        {
+            services.AddScoped<IOrderAppService, OrderAppService>();
+            services.AddScoped<IOrderRequestAppService, OrderRequestAppService>();
+            services.AddScoped<OrderRequestStatusChangedEventHandler>();
+            services.AddScoped<OrderStatusChangedEventHandler>();
```
Comment (Mmishaaa) [inline ServiceFinder.OrderService/ServiceFinder.OrderService.Application/ApplicationExtension.cs:51]: no interfaces decalred

File: ServiceFinder.OrderService/ServiceFinder.OrderService.Application/EventHandlers/OrderRequestStatusChangedEventHandler.cs (hunk 3 above)
Comment (Mmishaaa) [reply ServiceFinder.OrderService/ServiceFinder.OrderService.Application/EventHandlers/OrderRequestStatusChangedEventHandler.cs:7]: same

File: ServiceFinder.OrderService/ServiceFinder.OrderService.Application/EventHandlers/OrderStatusChangedEventHandler .cs [hunk 8]
```diff
@@ -0,0 +1,22 @@
+﻿using ServiceFinder.OrderService.Application.Messaging;
+using ServiceFinder.OrderService.Domain.Events;
+using System.Text.Json;
+
+namespace ServiceFinder.OrderService.Application.EventHandlers
+{
+    public class OrderStatusChangedEventHandler
```
Comment (Mmishaaa) [inline ServiceFinder.OrderService/ServiceFinder.OrderService.Application/EventHandlers/OrderStatusChangedEventHandler .cs:7]: interface missed

**Review by Mmishaaa — CHANGES_REQUESTED** [review]

File: ServiceFinder.OrderService/ServiceFinder.OrderService.Domain/Interfaces/IRepository.cs [hunk 9]
```diff
@@ -6,7 +6,7 @@ namespace ServiceFinder.OrderService.Domain.Interfaces
     public interface IRepository<T> where T : BaseModel
     {
         Task<T?> GetByIdAsync(Guid id, CancellationToken cancellationToken);
-        Task<PagedResult<T>> GetAllAsync(int pageNumber, int pageSize, CancellationToken cancellationToken);
+        PagedResult<T> GetAll(int pageNumber, int pageSize);
```
Comment (Mmishaaa) [inline ServiceFinder.OrderService/ServiceFinder.OrderService.Domain/Interfaces/IRepository.cs]: why not async

**Review by Mmishaaa — APPROVED** [review]

**Pull request merged**